lib/puppet/provider/gluster.rb

```diff
@@ -0,0 +1,101 @@
+class Puppet::Provider::Gluster < Puppet::Provider
+
+  # NOTE: This assumes that child providers define a `gluster` command:
+  #
+  # has_command(:gluster, 'gluster') do
+  #   environment :HOME => "/tmp"
+  # end
+
+  # Peer information
+
+  def self.parse_peer_status(output)
+    doc = REXML::Document.new(output)
+    xpath_match(doc, "/cliOutput/peerStatus/peer/hostname/text()")
+  end
+
+  def self.all_peers
+    parse_peer_status(gluster("peer", "status", "--xml"))
```

Comment:
There's the default console output which is much harder to parse. That's what I used before switching to `--xml` (which isn't documented anywhere I can find, but at least one other puppet module uses it) to simplify the parsing.